lib/waffle/storage/s3.ex

```diff
@@ -192,22 +192,41 @@ defmodule Waffle.Storage.S3 do
     end
   end
 
+  # If the file is a stream, send it to AWS as a multi-part upload
+  defp do_put(file = %Waffle.File{stream: file_stream}, {s3_bucket, s3_key, s3_options})
```

Comment:
It seems to share similar logic with the following function, can we re-use the common parts?

Reply:
Yes, I can refactor those 2 functions for better reuse.